lib/ai/model/output/streaming.go

```diff
@@ -0,0 +1,65 @@
+package output
+
+import (
+	"errors"
+	"io"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/sashabaranov/go-openai"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/lib/ai/tokens"
+)
+
+// StreamToDeltas converts an openai.CompletionStream into a channel of strings.
+// This channel can then be consumed manually to search for specific markers,
+// or directly converted into a StreamingMessage with NewStreamingMessage.
+func StreamToDeltas(stream *openai.ChatCompletionStream) chan string {
```

Comment:
Does this function need to be public? For me, it sounds like some internal helper that should not be called outside of the module.

Reply:
`lib/ai/model` and `lib/ai/model/tools` need to do this and are in separate package. I don't think we need to expose this outside of `lib/ai/model/...` though. `lib/ai/model` manually processes the deltas, while `lib/ai/model/tools` can create the streaming message directly, so we can't make a single `StreamtoStramingMessage()` function

Reply:
I can create a new `lib/api/model/utils` package if this helps